site/book/02-concepts/02-package-lifecycle.md

```diff
@@ -0,0 +1,53 @@
+In this section, we'll describe the typical workflows in kpt. We say "typical", because there is no
+single right way of using kpt. A user may choose to use some command but not another. This
+modularity is a key design principle. However, we still want to provide guidance on how the
+functionality could be used in a end-to-end real-world scenario.
+
+A workflow in kpt can be best modelled as performing some verbs on the noun _package_.
+
+For example, when consuming an upstream package, the initial workflow can look like this:
+
+![img](/static/images/lifecycle/flow1.svg)
+
+- **Get**: Using `kpt pkg get`
+- **Explore**: Using an editor or running commands such as `kpt pkg tree`
+- **Customize**: Manual edits or using `kpt fn eval`
+- **Render**: Using `kpt fn render`
+
+First, you get a package from upstream. Then, you explore the content of the package to understand
```

Comment:
I think we should separate publisher-package-lifecycle from consumer-package-lifecycle. 

`First, you get a package from upstream` -> this is a typical consumer workflow. Publishers may also init packages locally using `kpt pkg init` from scratch. We can remove the `publish` section for consumers.

Reply:
Added `flow5.svg` for creating a package from scratch. But note:

There are different dimensions:

1. I am using existing packages (single package or composing subpackages) vs creating new packages from scratch
2. I am deploying a package
3. I am publishing a package for others to use

These are not mutually exclusive and can be combined. For example:

a. I use an existing package, I deploy that package, and I also publish it for others. 
b. I create a new package, I don't deploy it, and I publish it for others.

Is flow a) `publisher-package-lifecycle` or `consumer-package-lifecycle`? I think it's both: You're consuming packages and then publishing them.

Instead of having binary flows for consumer and publisher, I broke down the e2e flow into smaller orthogonal sub-flows.